### PR TITLE
Adds support for clippy

### LINKED ISSUE #62: Clippy issue
Clippy is failing with the following warning, this is breaking the pre-commit hook.

```
    Checking astarte_sdk v0.1.0 (/home/matteo/projects/read/astarte-device-sdk-rust)                                                 
warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
   --> src/builder.rs:115:5                                                                                                          
    |                                                                                                                                
115 | /     pub fn database<'a, T: AstarteDatabase + 'static + Sync + Send>(                                                         
116 | |         &'a mut self,                                                                                                        
117 | |         database: T,                                                                                                         
118 | |     ) -> &'a mut Self {                                                                                                      
    | |_____________________^                                                                                                        
    |                                                                                                                                
    = note: `#[warn(clippy::needless_lifetimes)]` on by default                                                                      
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes               
```

### PR DESCRIPTION
**Pull request by sorru94**
Adds support for clippy by fixing all its warnings and by adding clippy to the CI.
Closes #62.

### CODE REVIEW

**Comment by sorru94** [comment]
One special mention to the warning `clippy::enum-variant-names`.
This warning is emitted when names of `enum` members end with the name of the `enum`.
The logic behind it is explained [here](https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names).

This seems reasonable, and as a consequence the members of the `enum` `Error` have been renamed accordingly removing the trailing `Error`.

**Comment by sorru94** [comment]
As a suggestion I would consider renaming the `Error` enumerator to something more explicative like `InterfaceError`.

File: src/lib.rs [hunk 1]
```diff
@@ -749,15 +749,15 @@ impl AstarteSdk {
             let interface_major = interfaces
                 .interfaces
                 .get(interface_name)
-                .ok_or_else(|| {
-                    AstarteError::InterfaceError(interface::Error::InterfaceNotFoundError)
-                })?
+                .ok_or(AstarteError::InterfaceError(
```
Comment (harlem88) [inline src/lib.rs:752]: What kind of alarms Clippy throw here? 
Arguments passed to ok_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use [ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else), which is lazily evaluated.

File: src/lib.rs [hunk 2]
```diff
@@ -749,15 +749,15 @@ impl AstarteSdk {
...
+                    interface::Error::InterfaceNotFound,
+                ))?
                 .get_version_major();
             let mapping = interfaces
                 .get_mapping(interface_name, interface_path)
-                .ok_or_else(|| {
-                    AstarteError::InterfaceError(interface::Error::MappingNotFoundError)
-                })?;
+                .ok_or(AstarteError::InterfaceError(
```
Comment (harlem88) [inline src/lib.rs:758]: same above

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (sorru94) [reply src/lib.rs:752]: The returned warning is: `clippy::unnecessary_lazy_evaluations`, description can be found [here](https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations).

The closure we are passing to `ok_or_else` is just the variant of an `enum` right?
Why would it make a difference if it's lazily or eagerly evaluated?

**Review by harlem88 — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**